components/brave_wallet/browser/internal/hd_key_sr25519_unittest.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_wallet/browser/internal/hd_key_sr25519.h"
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+TEST(HDKeySr25519, Basics) {
```

Comment:
Could you break it to separate test cases please?

Reply:
Done! The tests should look a lot nicer now.